File: abc301/c.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    string S, T;
    cin >> S >> T;

    int str_len = S.size(), s_at = 0, t_at = 0;
    string S_diff = "";
    regex re("[^atcoder@]");

    for (char c: S) {
        int index = T.find(c);

        if (c == '@') {
            S_diff += c;
            continue;
        }

        if (index != string::npos) {
            T.erase(T.begin()+index, T.begin()+(index+1));
        } else {
            S_diff += c;
        }
    }

    if (regex_search(S_diff, re) || regex_search(T, re)) {
        cout << "No" << endl;
        return 0;
    }

    int S_at = count(S_diff.begin(), S_diff.end(), '@'), T_at = count(T.begin(), T.end(), '@');

    if (S_at >= T.size() - T_at && T_at >= S_diff.size() - S_at) {
        cout << "Yes" << endl;
    } else {
        cout << "No" << endl;
    }
    
    return 0;
}

File: abc301/c.py
import re


def main():
    S = input()
    T = input()

    S_diff = ""

    for c in S:
        res = T.find(c)
        if c == '@':
            S_diff += c
            continue

        if res != -1:
            if res == 0:
                T = T[res+1:]
            else:
                T = T[:res] + T[res+1:]
        else:
            S_diff += c

    # print(S_diff)
    # print(T)

    if re.search(r"[^atcoder@]", S_diff) is not None or re.search(r"[^atcoder@]", T) is not None:
        print("No")
        return

    S_at = S_diff.count('@')
    T_at = T.count('@')
    S_len = len(S_diff)
    T_len = len(T)

    if S_at >= T_len - T_at and T_at >= S_len - S_at:
        print("Yes")
    else:
        print("No")


if __name__ == "__main__":
    main()
